google/cloud/dataproc_spark_connect/session.py

```diff
@@ -471,6 +475,43 @@ def generate_dataproc_session_id():
             )
             return f"sc-{timestamp}-{random_suffix}"
 
+    def __init__(
+        self,
+        connection: Union[str, DataprocChannelBuilder],
+        user_id: Optional[str] = None,
+    ):
+        """
+        Creates a new DataprocSparkSession for the Spark Connect interface.
+
+        Parameters
+        ----------
+        connection : str or :class:`DataprocChannelBuilder`
+            Connection string that is used to extract the connection parameters
+            and configure the GRPC connection. Or instance of ChannelBuilder /
+            DataprocChannelBuilder that creates GRPC connection.
+        user_id : str, optional
+            If not set, will default to the $USER environment. Defining the user
+            ID as part of the connection string takes precedence.
+        """
+
+        def print_spark_ui_url(client: SparkConnectClient):
```

Comment:
We create a nested method for this instead of just doing it directly in the `__init__` method body?

This method is only used here in `__init__`, and encapsulating the logic in a method neither reduces the amount of code nor makes it easier to read. On top of that, the name of the method doesn't seem to match what it does.

I would suggest inlining the method body into the enclosing `__init__` method and removing the `print_spark_ui_url` method definition entirely.